code/game/vault_door.dm

```diff
@@ -27,48 +30,70 @@
 /obj/structure/vaultdoor/proc/destroy()
 	icon_state = "empty"
 	set_opacity(0)
-	src.density = 0
-	destroyed = 1
+	src.density = FALSE
+	destroyed = TRUE
 
 /obj/structure/vaultdoor/proc/open()
-	is_busy = 1
+	is_busy = TRUE
 	flick("opening", src)
 	icon_state = "open"
 	playsound(loc, 'sound/f13machines/doorgear_open.ogg', 50, 0, 10)
 	sleep(30)
 	set_opacity(0)
-	src.density = 0
-	is_busy = 0
+	src.density = FALSE
+	is_busy = FALSE
+	is_open = TRUE
 
 /obj/structure/vaultdoor/proc/close()
-	is_busy = 1
+	is_busy = TRUE
 	flick("closing", src)
 	icon_state = "closed"
 	playsound(loc, 'sound/f13machines/doorgear_close.ogg', 50, 0, 10)
 	sleep(30)
 	set_opacity(1)
-	src.density = 1
-	is_busy = 0
+	src.density = TRUE
+	is_busy = FALSE
+	is_open = FALSE
 
 /obj/structure/vaultdoor/proc/vaultactivate()
 	if(destroyed)
-		usr << "<span class='warning'>[src] is broken</span>"
+		to_chat(usr, "<span class='warning'>[src] is broken</span>")
 		return
 	if(is_busy)
-		usr << "<span class='warning'>[src] is busy</span>"
+		to_chat(usr, "<span class='warning'>[src] is busy</span>")
 		return
-	if (density)
+	if(density)
 		open()
 		return
 	close()
 
-//� �� ���� ������������ ��� ������  - Google translate tells me that from Russian to english this is "Do not move your arms around this way." so dont.
+/obj/structure/vaultdoor/attackby(obj/item/I, mob/living/user, params)
+	add_fingerprint(user)
+	if(istype(I, /obj/item/weldingtool) && user.a_intent == INTENT_HELP)
+		if(obj_integrity < max_integrity)
+			if(!I.tool_start_check(user, amount=0))
+				return
+
+			to_chat(user, "<span class='notice'>You begin repairing [src]...</span>")
+			if(I.use_tool(src, user, 40, volume=50))
```

Comment:
I don't think there's any welder you can get normally that has a volume of 50 unless you somehow get a 80 volume industrial welding tool

Reply:
I think hacked lathes spit these out?

If this is an issue we can probably just make this a gradual thing. Like each weld uses 10 fuel and increases integrity (capped) by max_integrity/5 or something

Reply:
@ma44 and @LetterN 

I am fine with this change. Up to you lads if incremental repairs or something else should be used instead.